### PR TITLE
Flatten nested documentation into just 2nd-level pages, and revamp consistency and style

### LINKED ISSUE #289: Flatten documentation pages so that each endpoint is fully described in one page

### PR DESCRIPTION
**Pull request by cstephen**
Closes #289.

This PR revamps all API endpoint documentation in several ways:

- Separate (3rd-level) pages for point queries, area queries, etc. have been removed. The content of these pages has been folded into the main (2nd-level) page for the endpoint data category (e.g., "Geology"). For example, instead of clicking "Geology" -> "Point query" to see examples of how the Geology point query works, this information is now accessible by simply loading the "Geology" page itself.
- Because folding all point/area/etc. query text into parent category pages tended to bloat the pages and create an unstructured "wall of text" situation, I've made many formatting changes to add structure, rigidity, and visual differences. These include:
  - Listing all endpoints and corresponding example URLs in tables to make them stand out against explanatory text.
  - Using a consistent style for example output & structure. Before this, some endpoints used HTML lists to explain output structure, while others described the output structure in an HTML &lt;pre&gt; block. Now all endpoints use the &lt;pre&gt; block approach to explain output structure.
  - Using h3 and h4 tags in a more consistent manner between pages to separate endpoints vs. example output (h3 tags), and endpoint and output subcategories as needed (h4 tags).
  - Removing "Dimensions" documentation in the few places where it existed since this added a lot of bloat to pages, was not consistent between endpoint pages, and seemed redundant vs. what can be seen in the endpoint output on its own.
- All "/mmm" endpoints have been either moved into their respective data category pages (e.g., "Temperature and Precipitation") or have been moved into new 2nd-level data category pages that didn't exist before (Degree Days and Snowfall Equivalent).
- I found a few cases where URLs were being used as link names in "Source data" tables, causing the table to stretch far past the width of the page. In these cases, I've given these links shorter, more descriptive names so that the tables fit within the page.

To test, look at the page for each endpoint data category ("Climate Protection from Spruce Beetles", "Degree Days", "Design Index", etc.) and make sure the content makes sense and the endpoint URLs work and are accurately described.

### CODE REVIEW

**Review by BobTorgerson — APPROVED** [review]
This is great work Craig! Substantial clean-up of these documentation files!

I tested every documentation end point, along with every anchor link in the documentation pages to test both the endpoints and the backend data collection links.

I approve these changes!

**Review by brucecrevensten — CHANGES_REQUESTED** [review]
Intro checklist for all endpoints:
- Remove references to specific tools.
- What is this dataset?
- For modeled stuff:
    - What source dataset covers the historical portion?
    - What source dataset covers the future portion?
    - What models/scenarios/IPCC group is this from? i.e. “These data incorporate two models (x, y) and 1 scenario (z) from the (k) dataset.”
- For observed / other stuff: where’s the data from, in plain English?
- Define the timesteps available through the endpoint: is it pre-summarized?  Raw?  How/what?
- Define the (approx) spatial resolution.
- Link (if easy — to our own project page, like for ALFRESCO) inline to a source of basic info
- Blurb: “links to academic references and source datasets are included at the bottom of this page.”

Beetles:
- Needs academic paper
- Needs source data link
- Intro: this isn’t clipped to IEM

Degree days:
- [other ticket] refactor away from /mmm as top level
- Needs academic paper
- Needs source data link
- Needs intro improvement

Design index:
- [other ticket] refactor away from `historical|projected|hp|all`
- Needs intro improvement

Elevation: Good

Flam/Veg:
- Needs link to academic paper.  Intro is pretty good, but include link to https://uaf-snap.org/project/alfresco-and-habitat-research/ or similar

Landfast sea ice
- Improve intro per guidance above
- Add link to paper
- Add link to dataset (our Geoserver??)

Mean Annual Precipitation
- Check if the 771m is the correct future dataset (should be 2km?)
- Improve intro
- Add academic link

Permafrost:
- Do we remove all GIPL1?  Then we need to be clear about GIPL2.0
- Improve intro: “Query permafrost information for Alaska, including outputs from academic research on permafrost type and extent, as well as modeled ground temperatures from the GIPL2.0 model.” etc.
- Data source links / academic papers are probably OK here but double-check.  Dunno if we have a paper for GIPL2.

Sea Ice Concentration
- Include link to John’s research paper on this (linkable somewhere else too)

Snowfall Equivalent
- Slightly improve intro (just needs model scenario resolution etc)
- Add links to source data
- Add link to academic paper

Taspr
- Needs temporal resolution in intro
- Needs academic paper

Wet days per year
- [ticket] Consider moving under a /precipitation umbrella??  

Wildfire
- Improve intro
- Pull in the docs from Alaska Wildfire explorer tool + add the data sources section
- Add reference to academic paper for Alfresco
- Link to Peter’s lightning climatology
- Dunno about the NRT stuff in terms of academic papers tho

**Comment by cstephen** [comment]
I've updated the documentation for all endpoints with your suggestions @brucecrevensten, including:

- More substantial introductions for all endpoints, including mentions of historical and projected datasets that were used to produce the data, data resolution, and temporal ranges (as well as mentioning if data is accessible for individual years, summarized across eras, or both)
- Linking to academic papers for all endpoints except the beetle endpoint since the paper is in progress
- Linking to source datasets when they are available for download
- Trying to standardize the way introductions read and papers/dataset links are displayed across all datasets (as much as possible)

This PR is ready for a re-review when you have a chance.

**Comment by cstephen** [comment]
I'm going to go ahead and merge this PR since so many other documentation tickets will need to be based off of this work. If any of the work in this PR needs tweaked, we can take care of that with additional tickets/PRs.
